drivers/gpio/gpio_davinci.c

```diff
@@ -0,0 +1,185 @@
+				(struct gpio_davinci_regs *)config->port_base;
+
+	*value = regs->in_data;
+
+	return 0;
+}
+
+static int gpio_davinci_port_set_masked_raw(const struct device *dev,
+		gpio_port_pins_t mask, gpio_port_value_t value)
+{
+	struct gpio_davinci_config *config = (struct gpio_davinci_config *)dev->config;
+	volatile struct gpio_davinci_regs *regs =
+				(struct gpio_davinci_regs *)config->port_base;
+
+	regs->out_data = (regs->out_data & (~mask)) | (mask & value);
+
+	return 0;
+}
+
+static int gpio_davinci_port_set_bits_raw(const struct device *dev,
+									gpio_port_pins_t mask)
+{
+	struct gpio_davinci_config *config = (struct gpio_davinci_config *)dev->config;
+	volatile struct gpio_davinci_regs *regs =
+				(struct gpio_davinci_regs *)config->port_base;
+
+	regs->set_data |= mask;
+
+	return 0;
+}
+
+static int gpio_davinci_port_clear_bits_raw(const struct device *dev,
+									gpio_port_pins_t mask)
+{
+	struct gpio_davinci_config *config = (struct gpio_davinci_config *)dev->config;
+	volatile struct gpio_davinci_regs *regs =
+				(struct gpio_davinci_regs *)config->port_base;
+
+	regs->clr_data |= mask;
+
+	return 0;
+}
+
+static int gpio_davinci_port_toggle_bits(const struct device *dev,
+									gpio_port_pins_t mask)
+{
+	struct gpio_davinci_config *config = (struct gpio_davinci_config *)dev->config;
+	volatile struct gpio_davinci_regs *regs =
+				(struct gpio_davinci_regs *)config->port_base;
+
+	regs->out_data ^= mask;
+
+	return 0;
+}
+
+static const struct gpio_driver_api gpio_davinci_driver_api = {
+	.pin_configure = gpio_davinci_configure,
+	.port_get_raw = gpio_davinci_port_get_raw,
+	.port_set_masked_raw = gpio_davinci_port_set_masked_raw,
+	.port_set_bits_raw = gpio_davinci_port_set_bits_raw,
+	.port_clear_bits_raw = gpio_davinci_port_clear_bits_raw,
+	.port_toggle_bits = gpio_davinci_port_toggle_bits
+};
+
+static int gpio_davinci_init(const struct device *dev)
+{
+	return 0;
+}
+
+#define GPIO_J721E_INIT_FUNC(n)	\
+	static int gpio_davinci_##n##_init(const struct device *dev)	\
+	{	\
+		struct gpio_davinci_config *config = \
+				(struct gpio_davinci_config *)dev->config;	\
+		ARG_UNUSED(dev);	\
+		return 0;	\
+	}
+
+#define GPIO_J721E_DEVICE_INIT(n)	\
+	DEVICE_DT_INST_DEFINE(n, &gpio_davinci_##n##_init,	\
+		NULL, &gpio_davinci_##n##_data,	\
+		&gpio_davinci_##n##_config,	\
+		POST_KERNEL, CONFIG_GPIO_INIT_PRIORITY,	\
+		&api_funcs)
+
+#define GPIO_J721E_INIT(n)	\
+	static const struct gpio_davinci_config gpio_davinci_##n##_config = {	\
+		.common = {	\
+			.port_pin_mask = GPIO_PORT_PIN_MASK_FROM_DT_INST(n),	\
+		},	\
+		.port_base = DT_INST_REG_ADDR(n),	\
+		.port_num = n	\
+	};	\
+	\
+	static struct gpio_davinci_data gpio_davinci_##n##_data;	\
+	\
+	DEVICE_DT_INST_DEFINE(n,	\
+		&gpio_davinci_init,	\
+		NULL,	\
+		&gpio_davinci_##n##_data,	\
+		&gpio_davinci_##n##_config,	\
+		PRE_KERNEL_2,	\
+		CONFIG_GPIO_INIT_PRIORITY,	\
+		&gpio_davinci_driver_api);
+
+DT_INST_FOREACH_STATUS_OKAY(GPIO_J721E_INIT)
```

Comment:
Again, why `GPIO_J721E_INIT` ? Is it something specific to J721E? If yes then does it really belong here in the driver or somewhere more platform specific?

Reply:
updating to DAVINCI.